jwst/jump/jump_step.py

```diff
@@ -60,90 +68,115 @@ def process(self, step_input):
-                                      )
+                self.log.info('Maximum cores to use = %s', self.maximum_cores)
 
+            # Detect jumps using a copy of the input data model.
+            result = input_model.copy()
+            jump_data = self._setup_jump_data(result)
+            new_gdq, new_pdq, number_crs, number_extended_events, stddev = detect_jumps_data(jump_data)
+                
+            # Update the DQ arrays of the output model with the jump detection results
+            result.groupdq = new_gdq
+            result.pixeldq = new_pdq
+
+            # determine the number of groups with all pixels set to DO_NOT_USE
+            dnu_flag = dqflags.pixel["DO_NOT_USE"]
+            num_flagged_grps = 0
+            for integ in range(nints):
+                for grp in range(ngroups):
+                    if np.all(np.bitwise_and(result.groupdq[integ, grp, :, :], dnu_flag)):
+                        num_flagged_grps += 1
+
+            total_groups = nints * ngroups - num_flagged_grps - nints
+            if total_groups >= 1:
+                total_time = result.meta.exposure.group_time * total_groups
+                total_pixels = nrows * ncols
+
+                crs =  1000 * number_crs / (total_time * total_pixels)
+                result.meta.exposure.primary_cosmic_rays = crs
+
+                events =  1e6 * number_extended_events / (total_time * total_pixels)
+                result.meta.exposure.extended_emission_events = events
 
             tstop = time.time()
             self.log.info('The execution time in seconds: %f', tstop - tstart)
 
             result.meta.cal_step.jump = 'COMPLETE'
 
-            # Cleanup
-            del gain_model
-            del readnoise_model
-
         return result
+
+    def _setup_jump_data(self, result):
+        """
+        Create a JumpData instance to be used by STCAL jump.
+        """
+        # Get the gain and readnoise reference files
+        gain_filename = self.get_reference_file(result, 'gain')
+        self.log.info('Using GAIN reference file: %s', gain_filename)
+        readnoise_filename = self.get_reference_file(result,'readnoise')
+        self.log.info('Using READNOISE reference file: %s', readnoise_filename)
+
+        with datamodels.ReadnoiseModel(readnoise_filename) as rnoise_m, \
+                datamodels.GainModel(gain_filename) as gain_m:
+            # Get 2D gain and read noise values from their respective models
+            if reffile_utils.ref_matches_sci(result, gain_m):
+                gain_2d = gain_m.data
+            else:
+                log.info('Extracting gain subarray to match science data')
+                gain_2d = reffile_utils.get_subarray_data(result, gain_m)
+
+            if reffile_utils.ref_matches_sci(result, rnoise_m):
+                rnoise_2d = rnoise_m.data
+            else:
+                log.info('Extracting readnoise subarray to match science data')
```

Comment:
```suggestion
                self.log.info('Extracting readnoise subarray to match science data')
```
Gives an error for me as is; breaks the pipeline.